src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsService.java

```diff
@@ -351,40 +351,43 @@ protected void addFeedbackToResult(Result result, AbstractBuildResultNotificatio
     }
 
     /**
-     * Extracts the most helpful messages from the given test case.
+     * Extracts the most helpful message from the given test case.
      * @param testCase the test case information as received from Jenkins.
-     * @return a list of helpful messages that can be added to an automatic {@link Feedback}.
+     * @return the most helpful message that can be added to an automatic {@link Feedback}.
      */
-    private List<String> extractMessagesFromTestCase(final TestCaseDTO testCase) {
+    private Optional<String> extractMessageFromTestCase(final TestCaseDTO testCase) {
         var hasErrors = !CollectionUtils.isEmpty(testCase.getErrors());
         var hasFailures = !CollectionUtils.isEmpty(testCase.getFailures());
         var hasSuccessInfos = !CollectionUtils.isEmpty(testCase.getSuccessInfos());
         boolean successful = testCase.isSuccessful();
 
-        final List<String> feedbackMessages = new ArrayList<>();
+        String feedbackMessage;
 
         if (successful && hasSuccessInfos && testCase.getSuccessInfos().get(0).getMostInformativeMessage() != null) {
-            feedbackMessages.add(testCase.getSuccessInfos().get(0).getMostInformativeMessage());
+            feedbackMessage = testCase.getSuccessInfos().get(0).getMostInformativeMessage();
```

Comment:
I guess we could directly return here, right?

Reply:
That would duplicate the `return Optional.of` every time, but I can change it.